lib/src/default_transformer.dart

```diff
@@ -20,7 +20,14 @@ class DefaultTransformer extends Transformer {
   // TODO(nweiz): This should just take an AssetId when barback <0.13.0 support
   // is dropped.
   Future<bool> isPrimary(idOrAsset) {
-    var id = idOrAsset is AssetId ? idOrAsset : idOrAsset.id;
+    var id;
+    if (idOrAsset is Asset) {
+      id = (idOrAsset as Asset).id; // Only for old versions of barback.
+    } else if (idOrAsset is AssetId) {
+      id = idOrAsset;
+    } else {
+      id = new AssetId(null,null); // Not reached.
```

Comment:
how about keeping the original structure of the code, just adding a cast in the case of an asset:
```dart
var id = idOrAsset is AssetId ? idOrAsset : (idOrAsset as Asset).id;
```